proteus/LinearAlgebraTools.py

```diff
@@ -301,15 +413,85 @@ def scatter_reverse_add(self):
             self.proteus_array[:] = self.proteus_array[self.proteus2petsc_subdomain]
 
     def save(self, filename):
-        """Saves to disk using a PETSc binary viewer.
-        """
+        """Saves to disk using a PETSc binary viewer."""
         _petsc_view(self, filename)
 
-class ParMat_petsc4py(p4pyPETSc.Mat):
+class ParInfo_petsc4py:
     """
-    Parallel matrix based on petsc4py's wrappers for PETSc.
+    ARB - this class is experimental.  My idea is to store the
+    information need to constructor parallel vectors and matrices
+    here as static class values.  Then ParVec and ParMat can
+    use these values to create parallel objects later.
     """
-    def __init__(self,ghosted_csr_mat=None,par_bs=None,par_n=None,par_N=None,par_nghost=None,subdomain2global=None,blockVecType="simple",pde=None, par_nc=None, par_Nc=None, proteus_jacobian=None, nzval_proteus2petsc=None):
+
+    def __init__(self):
+        self.par_bs = None
+        self.par_n = None
+        self.par_n_lst = None
+        self.par_N = None
+        self.par_nghost = None
+        self.par_nghost_lst = None
+        self.petsc_subdomain2global_petsc = None
+        self.subdomain2global = None
+        self.proteus2petsc_subdomain = None
+        self.petsc2proteus_subdomain = None
+        self.nzval_proteus2petsc = None
+        self.dim = None
+        self.mixed = False
+
+    def print_info(cls):
+        import Comm
+        comm = Comm.get()
+        print 'comm.rank() = ' + `comm.rank()` + ' par_bs = ' + `cls.par_bs`
```

Comment:
Use logEvent, not print. If you want to log on all processors then use `-A, --logAllProcesses`. There are several reasons 1) this version of print goes away in Python 3, 2) if somebody accidentally uses this on a large multi-processor job it can easily fill up their stdio buffer, 3) we want the log to contain all relevant information